PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -2250,7 +2260,29 @@ void ScintillaEditView::restyleBuffer()
 
 void ScintillaEditView::styleChange()
 {
-	defineDocType(_currentBuffer->getLangType());
+	const bool isSameLangType = _prevBuffer != nullptr && ((_prevBuffer == _currentBuffer) || (_prevBuffer->getLangType() == _currentBuffer->getLangType()));
+	const int currentLangInt = static_cast<int>(_currentBuffer->getLangType());
+	const bool isFirstActiveBuffer = (_currentBuffer->getLastLangType() != currentLangInt) || (_currentBuffer->isUntitled());
+
+	if (isFirstActiveBuffer)
+	{
+		defineDocType(_currentBuffer->getLangType());
+	}
+	else if (!isSameLangType)  // When entering the tab for the second or more times
+	{
+		Document prevDoc = execute(SCI_GETDOCPOINTER);
+		execute(SCI_SETMODEVENTMASK, MODEVENTMASK_OFF);
+		execute(SCI_SETDOCPOINTER, 0, getBlankDocument());
+		execute(SCI_SETMODEVENTMASK, MODEVENTMASK_ON);
+
+		defineDocType(_currentBuffer->getLangType());
+
+		execute(SCI_SETMODEVENTMASK, MODEVENTMASK_OFF);
+		execute(SCI_SETDOCPOINTER, 0, prevDoc);
+		execute(SCI_SETMODEVENTMASK, MODEVENTMASK_ON);
+	}
+
+	_currentBuffer->setLastLangType(currentLangInt);

```

Comment:
`void ScintillaEditView::styleChange()` is called when user stays on the same document and is adding/deleting/modifying UDL element on UDL dialog. The aim of calling the fuction is updating the visual effect of document while the modification. 
As a result, the above code is not necessary.